pkg/pillar/cmd/zedagent/parseconfig.go

```diff
@@ -551,9 +551,13 @@ func parseAppInstanceConfig(config *zconfig.EdgeDevConfig,
 		for _, adapter := range cfgApp.Adapters {
 			log.Tracef("Processing adapter type %d name %s",
 				adapter.Type, adapter.Name)
+			hwaddr, err := net.ParseMAC(adapter.EthVf.Mac)
+			if err != nil {
+				log.Errorf("Failed to parse hardware address for adapter: %s", adapter.Name)
```

Comment:
Shouldn't this set an error in the AppInstanceStatus in addition to logging it?
Also, include the err.

Reply:
I could not find reference, could you show snippet?

Reply:
Just realized parseconfig has this for Network instances and networks but not for app instances.

That pattern includes an ErrorInfo in the *Config struct and parseconfig doing e.g.,
				errStr := fmt.Sprintf("Network Instance %s parameter parse failed: %s",
					networkInstanceConfig.Key(), err)
				log.Error(errStr)
				networkInstanceConfig.SetErrorNow(errStr)
				// Proceed to send error back to controller

and then when processing that *Config there is a check e.g., in zedrouter/networkinstance.go checking config.HasError() and then setting that error in *Status.
This complexity is needed since zedagent doesn't own the *Status content.